File: Source/Fantasy.Jobs/When.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.XSerialization;

namespace Fantasy.Jobs
{
    [Instruction]
    [XSerializable("when", NamespaceUri=Consts.XNamespaceURI )] 
    public class When : Sequence
    {
        [XAttribute("condition")]
        public string Condition { get; set; }

        public override void Execute()
        {
            this.ExecuteSequence(); 
        }
    }
}

File: Fantasy.Jobs/src/fantasy/jobs/When.java
package fantasy.jobs;

import fantasy.xserialization.*;

@Instruction
@XSerializable(name = "when", namespaceUri=Consts.XNamespaceURI)
public class When extends Sequence
{
	@XAttribute(name ="condition")
	public String Condition = null;

	@Override
	public void Execute() throws Exception
	{
		this.ExecuteSequence();
	}
}